File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/IEventInbox.cs
using System;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;

namespace Volo.Abp.EventBus.Distributed;

public interface IEventInbox
{
    void EnqueueAsync(IncomingEventInfo incomingEvent);

    Task<List<IncomingEventInfo>> GetWaitingEventsAsync(int maxCount, CancellationToken cancellationToken = default);

    void MarkAsProcessedAsync(Guid id);

    Task<bool> ExistsByMessageIdAsync(String messageId);

    void DeleteOldEventsAsync();
}

File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/IEventInbox.java
package com.ueueo.eventbus.distributed;

import com.ueueo.ID;
import com.ueueo.threading.CancellationToken;

import java.util.List;

public interface IEventInbox {

    void enqueue(IncomingEventInfo incomingEvent);

    List<IncomingEventInfo> getWaitingEventsAsyn(int maxCount, CancellationToken cancellationToken);

    void markAsProcessed(ID id);

    Boolean existsByMessageId(String messageId);

    void deleteOldEvents();
}
